smoke-tests/src/test/java/com/example/javaagent/smoketest/DynamicConfigSmokeTest.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.example.javaagent.smoketest;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.protobuf.ByteString;
+import io.opentelemetry.proto.collector.trace.v1.ExportTraceServiceRequest;
+import io.opentelemetry.proto.trace.v1.Span;
+import java.util.List;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class DynamicConfigSmokeTest extends TestAppSmokeTest {
+
+  @BeforeAll
+  public static void start() {
+    startTestApp(
+        (container) -> {
+          container.addEnv(
+              "OTEL_INSTRUMENTATION_METHODS_INCLUDE",
+              "co.elastic.otel.test.DynamicConfigController[flipSending]");
+          container.addEnv(
+              "ELASTIC_OTEL_JAVA_EXPERIMENTAL_DISABLE_INSTRUMENTATIONS_CHECKER", "true");
+          container.addEnv(
+              "ELASTIC_OTEL_JAVA_EXPERIMENTAL_DISABLE_INSTRUMENTATIONS_CHECKER_INTERVAL_MS", "300");
+          container.addEnv("OTEL_JAVAAGENT_DEBUG", "true");
+        });
+  }
+
+  @AfterAll
+  public static void end() {
+    stopApp();
+  }
+
+  @AfterEach
+  public void endTest() throws InterruptedException {
+    doRequest(getUrl("/dynamicconfig/reset"), okResponseBody("reset"));
+    Thread.sleep(500L); // give the reset time to be applied
```

Comment:
there was a reason which is that I prefer the smoke tests and find the integration tests non-transparent in how they work. Though I take your point they are easier to maintain. And these tests **are** actually integration testst. I've done an initial pass and found a bug the integration test identified that the smoke tests didn't, so I'll map them over

Reply:
per discussion (the integration test is unable to handle the exporter wrapping correctly because it has a custom exporter that doesn't go through configuration), staying with the smoke test